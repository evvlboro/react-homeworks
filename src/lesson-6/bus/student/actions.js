import { types } from './types';

export const userActions = {
  fillProfile: (profile) => {
    return {
      type: types.STUDENT_FILL,
      payload: profile,
    }
  }
};
