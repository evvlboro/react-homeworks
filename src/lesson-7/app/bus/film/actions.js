// API
import { api } from '../../../api';

// Types
import { types } from './types';

export const filmActions = Object.freeze({
  // Sync
  startFetching: () => {
    return {
      type: types.FILM_START_FETCHING
    };
  },
  stopFetching: () => {
    return {
      type: types.FILM_STOP_FETCHING
    };
  },
  fill: (payload) => {
    return {
      type: types.FILM_FILL,
      payload
    };
  },
  setFetchingError: (error) => {
    return {
      type: types.FILM_SET_FETCHING_ERROR,
      error: true,
      payload: error
    };
  },

  // Async
  fetchAsync: (id) => async (dispatch) => {
    dispatch(filmActions.startFetching());

    const response = await api.film.fetch(id);

    if (response.status === 200) {
      const results = await response.json();

      dispatch(filmActions.fill(results));
    } else {
      const error = {
        status: response.status
      };

      dispatch(filmActions.setFetchingError(error));
    }

    dispatch(filmActions.stopFetching());
  }

});
