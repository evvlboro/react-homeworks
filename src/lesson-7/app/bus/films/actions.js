// API
import { api } from '../../../api';

// Types
import { types } from './types';

export const filmsActions = Object.freeze({
  // Sync
  startFetching: () => {
    return {
      type: types.FILMS_START_FETCHING
    };
  },
  stopFetching: () => {
    return {
      type: types.FILMS_STOP_FETCHING
    };
  },
  fill: (payload) => {
    return {
      type: types.FILMS_FILL,
      payload
    };
  },
  setFetchingError: (error) => {
    return {
      type: types.FILMS_SET_FETCHING_ERROR,
      error: true,
      payload: error
    };
  },

  // Async
  fetchAsync: () => async (dispatch) => {
    dispatch(filmsActions.startFetching());

    const response = await api.films.fetch();

    if (response.status === 200) {
      const {results} = await response.json();

      dispatch(filmsActions.fill(results));
    } else {
      const error = {
        status: response.status
      };

      dispatch(filmsActions.setFetchingError(error));
    }

    dispatch(filmsActions.stopFetching());
  }

});
