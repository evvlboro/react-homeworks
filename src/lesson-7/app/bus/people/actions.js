// API
import { api } from '../../../api';

// Types
import { types } from './types';

export const peopleActions = Object.freeze({
  // Sync
  startFetching: () => {
    return {
      type: types.PEOPLE_START_FETCHING
    };
  },
  stopFetching: () => {
    return {
      type: types.PEOPLE_STOP_FETCHING
    };
  },
  fill: (payload) => {
    return {
      type: types.PEOPLE_FILL,
      payload
    };
  },
  setFetchingError: (error) => {
    return {
      type: types.PEOPLE_SET_FETCHING_ERROR,
      error: true,
      payload: error
    };
  },

  // Async
  fetchAsync: () => async (dispatch) => {
    dispatch(peopleActions.startFetching());

    const response = await api.people.fetch();

    if (response.status === 200) {
      const {results} = await response.json();

      dispatch(peopleActions.fill(results));
    } else {
      const error = {
        status: response.status
      };

      dispatch(peopleActions.setFetchingError(error));
    }

    dispatch(peopleActions.stopFetching());
  }

});
