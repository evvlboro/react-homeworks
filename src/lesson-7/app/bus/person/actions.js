// API
import { api } from '../../../api';

// Types
import { types } from './types';

export const personActions = Object.freeze({
  // Sync
  startFetching: () => {
    return {
      type: types.PERSON_START_FETCHING
    };
  },
  stopFetching: () => {
    return {
      type: types.PERSON_STOP_FETCHING
    };
  },
  fill: (payload) => {
    return {
      type: types.PERSON_FILL,
      payload
    };
  },
  setFetchingError: (error) => {
    return {
      type: types.PERSON_SET_FETCHING_ERROR,
      error: true,
      payload: error
    };
  },

  // Async
  fetchAsync: (id) => async (dispatch) => {
    dispatch(personActions.startFetching());

    const response = await api.person.fetch(id);

    if (response.status === 200) {
      const results = await response.json();

      dispatch(personActions.fill(results));
    } else {
      const error = {
        status: response.status
      };

      dispatch(personActions.setFetchingError(error));
    }

    dispatch(personActions.stopFetching());
  }

});
