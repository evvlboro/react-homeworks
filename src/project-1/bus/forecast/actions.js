// Instruments
import * as config from './config';

export const fetchForecastSuccessAction = (forecast) => ({
    type:    config.FETCH_MANY.SUCCESS,
    payload: forecast,
});
