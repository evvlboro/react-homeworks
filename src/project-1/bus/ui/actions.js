// Instruments
import * as config from './config';

export const setActiveDayAction = (payload) => ({
    type: config.SET_ACTIVE_DAY,
    payload,
});

export const setFilterStateAction = (fieldName, value) => ({
    type: config.SET_FILTER_STATE,
    payload: {
        fieldName,
        value
    },
});

export const setTogglerStateAction = (fieldName) => ({
    type: config.SET_TOGGLER_STATE,
    payload: fieldName,
});

export const filterForecastAction = (isReset) => ({
    type: config.FILTER_FORECAST,
    payload: isReset
});
