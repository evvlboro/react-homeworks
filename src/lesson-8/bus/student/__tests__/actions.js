import { userActions } from '../actions';

describe("User actions", () => {
  it('should fill student', () => {
    expect(userActions.fillProfile(
      {
        firstName: 'Уилл',
        surname: 'Тернер',
        age: '19',
        email: 'blablabla@gmail.com',
        sex: 'male',
        speciality: 'writer',
      }
    )).toMatchSnapshot();
  });
})
