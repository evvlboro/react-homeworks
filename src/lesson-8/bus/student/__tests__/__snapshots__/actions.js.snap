// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User actions should fill student 1`] = `
Object {
  "payload": Object {
    "age": "19",
    "email": "blablabla@gmail.com",
    "firstName": "Уилл",
    "sex": "male",
    "speciality": "writer",
    "surname": "Тернер",
  },
  "type": "USER_FILL",
}
`;
